app/components/userProfile/validationSchema.ts

```diff
@@ -0,0 +1,30 @@
+import { z, ZodType } from 'zod';
+
+import { invalid_file_size } from '@/common/formValidation';
+import { UserSession } from '@/common/types';
+import { clientConfig } from '@/config/client';
+
+const MAX_FILE_SIZE = clientConfig.NEXT_PUBLIC_BODY_SIZE_LIMIT;
+
+const userSessionSchema = z.object({
+  providerId: z.string(),
+  provider: z.string(),
+  name: z.string(),
+  username: z.string().optional(),
+  role: z.string().optional(),
+  department: z.string().optional(),
+  email: z.string().email(),
+}) satisfies ZodType<UserSession>;
+
+export const handleUpdateUserSession = userSessionSchema.extend({
+  image: z
+    .union([
+      z.string(),
+      z.number(),
+      z.instanceof(File).refine((file) => file.size < MAX_FILE_SIZE, invalid_file_size),
+      z.any(),
+    ])
+    .optional(),
```

Comment:
since we only allow formData as the input for the image anyway, we can omit the union and use just `File`
```suggestion
  image: z.instanceof(File).refine((file) => file.size < MAX_FILE_SIZE, invalid_file_size).optional(),
```

Reply:
I think whenever frontend and backend require different types for validation we should separate them so my vote goes to the second option